### PR TITLE
chore: bump docker dependency to v20.10.5

### PR DESCRIPTION
**Pull request by mavogel**
In #102 we need to update dependencies due to broken windows build. So we split this out
- https://github.com/golang/go/issues/34610
- https://github.com/moby/moby/pull/40021

### Steps
- [x] update dependencies
- [x] adapt to new API (update will be done in #94)

### CODE REVIEW

File: website/docs/r/service.html.markdown [hunk 1]
```diff
@@ -454,9 +444,6 @@ the extra mount mappings for the container. Each `configs` is a reference to a s
 * `limits` - (Optional, list of strings) Describes the resources which can be advertised by a node and requested by a task.
   * `nano_cpus` (Optional, int) CPU shares in units of 1/1e9 (or 10^-9) of the CPU. Should be at least 1000000
   * `memory_bytes` (Optional, int) The amount of memory in bytes the container allocates
-  * `generic_resources` (Optional, map) User-defined resources can be either Integer resources (e.g, SSD=3) or String resources (e.g, GPU=UUID1)
-    * `named_resources_spec` (Optional, set of string) The String resources, delimited by `=`
-    * `discrete_resources_spec` (Optional, set of string) The Integer resources, delimited by `=`
```
Comment (suzuki-shunsuke) [inline website/docs/r/service.html.markdown:459]: This PR includes breaking changes, right?

Comment (mavogel) [reply website/docs/r/service.html.markdown:459]: Yes, it will include BCs and I guess #102 as well. So we will bump a major.

Comment (suzuki-shunsuke) [reply website/docs/r/service.html.markdown:459]: I see.

Comment (suzuki-shunsuke) [reply website/docs/r/service.html.markdown:459]: 📝 I don't know the feature `generic_resources` well. I'll investigate later. 

* https://docs.docker.com/engine/reference/commandline/service_create/#create-services-requesting-generic-resources
* https://pkg.go.dev/github.com/docker/docker@v20.10.5+incompatible/api/types/swarm#GenericResource
* https://pkg.go.dev/github.com/docker/docker@v20.10.5+incompatible/api/types/swarm#ResourceRequirements

**Review by suzuki-shunsuke — APPROVED** [review]

**Pull request merged**